vlm/src/main/scala/geotrellis/contrib/vlm/gdal/GDALResampleRasterSource.scala

```diff
@@ -8,59 +8,53 @@ import geotrellis.proj4.CRS
 import geotrellis.raster.reproject.Reproject
 
 import cats.implicits._
-import org.gdal.gdal.{Dataset, gdal}
 
 case class GDALResampleRasterSource(
   uri: String,
   resampleGrid: ResampleGrid,
-  method: ResampleMethod = NearestNeighbor
-) extends GDALBaseRasterSource { self =>
-  @transient lazy val dataset: Dataset = {
-    val baseDataset = self.baseDataset
-    val warpOptions =
-      resampleGrid match {
-        case Dimensions(cols, rows) =>
-          GDALWarpOptions(
-            dimensions = (cols, rows).some,
-            resampleMethod = method.some
-          )
-        case _ =>
-          val rasterExtent: RasterExtent = {
-            val colsLong: Long = baseDataset.getRasterXSize
-            val rowsLong: Long = baseDataset.getRasterYSize
-
-            val cols: Int = colsLong.toInt
-            val rows: Int = rowsLong.toInt
-
-            val geoTransform: Array[Double] = baseDataset.GetGeoTransform
-
-            val xmin: Double = geoTransform(0)
-            val ymin: Double = geoTransform(3) + geoTransform(5) * rows
-            val xmax: Double = geoTransform(0) + geoTransform(1) * cols
-            val ymax: Double = geoTransform(3)
-
-            RasterExtent(Extent(xmin, ymin, xmax, ymax), cols, rows)
-          }
-          val targetRasterExtent = resampleGrid(rasterExtent)
-          GDALWarpOptions(
-            cellSize = targetRasterExtent.cellSize.some,
-            alignTargetPixels = false,
-            resampleMethod = method.some
-          )
-      }
+  method: ResampleMethod = NearestNeighbor,
+  baseWarpList: List[GDALWarpOptions] = Nil
+) extends GDALBaseRasterSource {
+  lazy val warpOptions: GDALWarpOptions = {
+    val res = resampleGrid match {
+      case Dimensions(cols, rows) =>
+        GDALWarpOptions(
+          dimensions = (cols, rows).some,
+          resampleMethod = method.some
+        )
```

Comment:
Should there be a `case` that matches for [TargetRegion](https://github.com/geotrellis/geotrellis-contrib/blob/master/vlm/src/main/scala/geotrellis/contrib/vlm/ResampleGrid.scala#L35) here? Since we'll just be using the `RasterExtent` that object has anyway.

Reply:
No, TargetRegion.apply returns the initial RasterExtent. I decided not to copy-paste the code.